### PR TITLE
feat(Expander): Add WinUI Expander

### PR DESCRIPTION
**Pull request by robloo**
GitHub Issue (If applicable): #

Relates to #4425

This will not close the issue as it's only an initial port. Additional changes will be made in WinUI that need to also be copied over before release.

## PR Type

What kind of change does this PR introduce?

- Feature

## What is the current behavior?

No support for Microsoft.UI.Xaml.Controls.Expander

## What is the new behavior?

Support for Microsoft.UI.Xaml.Controls.Expander

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Source code derived from WinUI (licensed MIT). https://github.com/microsoft/microsoft-ui-xaml

Internal Issue (If applicable):

### CODE REVIEW

**Comment by robloo** [comment]
@MartinZikmund The code is in now if you want to start looking at it. 

 * While the expander appears on the test page you cannot expand/collapse it. I'll debug this more later but its an issue with the visual state manager. If you are feeling motivated to look into it I won't be upset :)
 * The AutomationPeer currently has a lot of disabled functionality.

**Comment by jeromelaban** [comment]
@robloo did you find any additional insights for this PR ?

**Comment by robloo** [comment]
@jeromelaban This port was completed except for the bugs which appear to be in Uno. I havn't spent the time to address them as the WinUI version wasn't released. It seems to be only a basic preview and I have no visibility on when it might be finished upstream.

If you don't have any concerns about enabling a preview control in Uno (that could change at any time) I can look at finishing this. It is a showstopper for anyone using expander right now as it crashes Uno entirely.

**Comment by MartinZikmund** [comment]
@robloo Hey! If you can get the PR to be ready for review (even though some things may not work due to Uno bugs), it would be perfect 🚀 . We will try try to fix the underlying issues after the fact.

File: src/Uno.UI/Microsoft/UI/Xaml/Controls/Expander/Expander.cs [hunk 1]
```diff
@@ -3,10 +3,11 @@
 using Windows.UI.Xaml.Automation.Peers;
 using Windows.UI.Xaml.Controls;
 using Windows.UI.Xaml.Controls.Primitives;
+using Windows.UI.Xaml.Markup;
 
 namespace Microsoft.UI.Xaml.Controls
 {
-	public partial class Expander : Control
+	public partial class Expander : ContentControl
```
Comment (robloo) [inline src/Uno.UI/Microsoft/UI/Xaml/Controls/Expander/Expander.cs:10]: Dumb mistake on my part. Hopefully didn't take too long to track down.

**Review by robloo — COMMENTED** [review]
Thanks for making the changes to get this in shape. I'll do a comparison with latest WinUI source tomorrow if you haven't already. Otherwise I'm assuming this is all set - I'll test it myself tomorrow.

**Review by jeromelaban — APPROVED** [review]

File: src/Uno.UI/Microsoft/UI/Xaml/Controls/Expander/Expander.cs (hunk 1 above)
Comment (MartinZikmund) [reply src/Uno.UI/Microsoft/UI/Xaml/Controls/Expander/Expander.cs:10]: Happens 😀 . It was a bit tricky, but @carldebilly helped me notice it 😀 .

**Comment by MartinZikmund** [comment]
@robloo There were no changes since your port, except for the new 2.6 visual styles, which we don't yet support, so it should be all good :-) .

File: src/Uno.UI/Microsoft/UI/Xaml/Controls/Expander/Expander.cs (hunk 1 above)
Comment (robloo) [reply src/Uno.UI/Microsoft/UI/Xaml/Controls/Expander/Expander.cs:10]: @MartinZikmund Well, sorry about that!

Comment (MartinZikmund) [reply src/Uno.UI/Microsoft/UI/Xaml/Controls/Expander/Expander.cs:10]: No problem at all 😀 !

**Comment by robloo** [comment]
@MartinZikmund  Thanks for getting this into shape. Really helped a lot!

It looks like you added some of the missing AutomationPeer functionality for expander as well `IExpandCollapseProvider`. However, I left quite a bit of the code commented out. Does Uno even support automation? If so, adding back the commented out code in `ExpanderAutomationPeer` might be a good idea. Otherwise, everything looks good to me.

**Review by carldebilly — APPROVED** [review]

**Pull request merged**